src/helics/common/AsioServiceManager.cpp

```diff
@@ -147,32 +144,31 @@ AsioServiceManager::LoopHandle AsioServiceManager::runServiceLoop (const std::st
     throw (std::invalid_argument ("the service name specified was not available"));
 }
 
-
-AsioServiceManager::LoopHandle AsioServiceManager::startServiceLoop () 
+AsioServiceManager::LoopHandle AsioServiceManager::startServiceLoop ()
 {
-    ++runCounter; //atomic
+    ++runCounter;  // atomic
 
     bool exp = false;
     if (running.compare_exchange_strong (exp, true))
     {
         auto ptr = shared_from_this ();
-        std::packaged_task<void()> serviceTask ([ptr=std::move(ptr)]() { serviceProcessingLoop (ptr); });
-     //   std::cout << "run Service loop " << runCounter << "\n";
+        std::packaged_task<void()> serviceTask ([ptr = std::move (ptr)]() { serviceProcessingLoop (ptr); });
```

Comment:
This is a closure, correct? Why wouldn't you `std::move` the `ptr` in the closure body? My understanding is that you rvalue cast in the arguments for the closure, right? I thought it would have been written as below.

```
std::packaged_task<void()> serviceTask ([ptr = std::move (ptr)]() { serviceProcessingLoop (std::move(ptr)); });
```

It's probably just an extra copy, so it's fine. I'm just curious.

Reply:
There's a bunch of other places in the code that I see a similar change required.